config/src/main/java/com/redhat/lightblue/config/LightblueFactory.java

```diff
@@ -173,6 +174,8 @@ private synchronized void initializeMetadata() throws IOException, ClassNotFound
             getJsonTranslator().setValidation(EntityInfo.class, cfg.isValidateRequests());
 
             metadata = cfg.createMetadata(datasources, getJSONParser(), this);
+            initializeFactory(); //re enforce the factory initialization
```

Comment:
What if we forced the factory to be passed in? The getMetadata method could simply pass in getFactory().